packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
Actually the error is most likely to occur from session pooling, say they had no sessions free and configured the pool to throw an error in that specific event.

I feel like the example you are proposing says to me "do not support promises in `runTransaction`".  I guess my decision to agree or disagree heavily depends on whether or not the expectation is to inform the user whether or not a Transaction was aborted while still retrying. I assumed that since we're retrying, we would only inform the user when we either got the final failure or a success (like any other API call retry strategy).